### PR TITLE
Add inline documentation for Type and DataType.

### LINKED ISSUE #23655: Missing inline documentation for Type and DataType
As discussed in the discourse (https://discourse.julialang.org/t/how-to-understand-type/5810), although it maybe lengthy to explain intended usage of `Type`, some basic description in inline docstring would be helpful.
```julia
help?> Type
search: Type typeof TypeVar TypeName typemin typemax TypedSlot typejoin

  No documentation found.

  Core.Type is of type UnionAll.

  Summary:

  struct UnionAll <: Type{T}

  Fields:

  var  :: TypeVar
  body :: Any

help?> DataType
search: DataType

  No documentation found.

  Summary:

  mutable struct DataType <: Type{T}

  Fields:

  name             :: TypeName
  super            :: DataType
  parameters       :: SimpleVector
  types            :: SimpleVector
  instance         :: Any
  layout           :: Ptr{Void}
  size             :: Int32
  ninitialized     :: Int32
  uid              :: Int32
  abstract         :: Bool
  mutable          :: Bool
  llvm::StructType :: Ptr{Void}
  llvm::DIType     :: Ptr{Void}
  depth            :: Int32
  hasfreetypevars  :: Bool
  isleaftype       :: Bool
```

### PR DESCRIPTION
**Pull request by phrb**
Closes #23655.

### CODE REVIEW

File: base/docs/basedocs.jl [hunk 1]
```diff
@@ -821,6 +821,76 @@ The singleton type containing only the value `Union{}`.
 """
 Core.TypeofBottom
 
+"""
+    Core.Type{T} where T
+
+`Core.Type` is an abstract type which has all type objects as its instances.
+The only instance of the singleton type `Core.Type{T} where T` is the object
```
Comment (JeffBezanson) [inline base/docs/basedocs.jl]: This should say `Type{T}`, so the `T` is free.

**Comment by ViralBShah** [comment]
I think this would be nice to get in. @phrb, if you are up for it, can you rebase to master and trim the PR examples to make it concise?

**Comment by ViralBShah** [comment]
The failures need investigating...

**Comment by ViralBShah** [comment]
Sorry I had a whitespace bug committed. Can you rebase on master? There are some other issues that need to be addressed too.

**Comment by phrb** [comment]
I've rebased, trimmed examples, and corrected type annotations.

File: base/docs/basedocs.jl (hunk 1 above)
Comment (mbauman) [reply base/docs/basedocs.jl:1157]:
```suggestion
The only instance of the singleton type `Core.Type{T}` is the object
```

**Review by mbauman — APPROVED** [review]

**Pull request merged**